include/gui/application.hpp

```diff
@@ -180,23 +179,38 @@ namespace Toolbox {
 
     class FileDialog {
     public:
-        static FileDialog* Instance() {
+        FileDialog() = default;
+        ~FileDialog() {
+            if (m_thread_initialized) {
+                m_thread.join();
+            }
+        }
+
+        static FileDialog *Instance() {
             static FileDialog _instance;
             return &_instance;
         }
-        void OpenDialog(std::filesystem::path starting_path,
-                        GLFWwindow* parent_window,
+        void OpenDialog(std::filesystem::path starting_path, GLFWwindow *parent_window,
                         bool is_directory = false,
-                        std::optional<std::vector<std::pair<
-                        const char*, const char*>>> maybe_filters = std::nullopt);
+                        std::optional<std::vector<std::pair<std::string, std::string>>>
+                            maybe_filters = std::nullopt);
         bool isAlreadyOpen() { return m_thread_running; }
-        bool isDone() { return !m_thread_running && not m_closed; }
+        bool isDone() { return !m_thread_running && !m_closed && m_thread_initialized; }
         bool isOk() { return m_result == NFD_OKAY; }
         std::filesystem::path GetFilenameResult() { return m_selected_path; }
-        void Close() { m_closed = true;}
+        void Close() { m_closed = true; }
+
     private:
+#ifdef TOOLBOX_PLATFORM_WINDOWS
+        std::wstring m_starting_path;
+        std::vector<std::pair<std::wstring, std::wstring>> m_filters;
+#else
+        std::string m_starting_path;
+        std::vector<std::pair<std::string, std::string>> m_filters;
+#endif
+
```

Comment:
Why store these on the object? They're set with every open call, and we don't seem to use them outside that block, so it seems like they could just be local variables